### PR TITLE
Updates for Convective Transition POD for using python3_base environment

### PR DESCRIPTION
**Pull request by yihungkuo**
**Description**
Updates for Convective Transition POD so that it can run under the python3_base environment.
The required dependencies (numba and networkx) are added to the python3_base environment (together with jupyterlab). The YAML file for the convective_transition_diag environment is deleted.

**How Has This Been Tested?**
The updates were made to the latest GFDL develop branch. After updates, the package is re-downloaded from the GitHub website, and then the base and python3_base environments are re-installed. The setup of Conda environments follows README.md (using Miniconda2) with the updated YAML file for python3_base. Convective Transition POD is then tested. The POD correctly produced the webpages and figures (passed visual inspection).

Relevant details:
  OS: Linux Mint 20
  CPU:  Intel i7-10710U
  Memory: 40G
  Wall-clock time: ~20s
  CASENAME tested: QBOi.EXP1.AMIP.001

Remark:  Convective Transition POD has been included in the previous releases of the MDTF package. Thus the test here is only to confirm the updates for working under the python3_base environment.

**Checklist:**
- [x] I have reviewed my own code to ensure that if follows the [POD development guidelines](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/dev_guidelines.html)
- [x] The script are written in Python 3.6 or above (preferred; required if funded by a CPO grant), NCL, or R
- [x] All of my scripts are in the diagnostics/ subdirectory, and include a main_driver script template html, and settings.jsonc
- [x] The main_driver script header has all of the information in the POD documentation, excluding the "More about this diagnostic" section
- [x] The POD directory and html template have the same short name as my POD
- [x] The html template has a 1-paragraph synopsis of the POD and links to the main documentation
- [x] If applicable, I've added a .yml file to src/conda, and my environment builds with conda_env_setup.sh 
- [x] The POD scripts do not access the internet or networked resources
- [x] I have commented the code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation in the POD's doc/ subdirectory
- [x] I have created the directory input_data/obs_data/[pod short name]
- [x] My code is portable; it uses [MDTF environment variables](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/ref_envvars.html), and does not contain hard-coded file or directory paths
- [x] I have added information about the raw data files to the documentation
- [x] I have provided the code to generate digested data files from raw data files
- [x] Each digested data file generated by the script contains numerical data (no figures), and is 3 GB or less in size
- [x] The repository contains no extra test scripts or data files
- [x] My branch is up-to-date with the NOAA-GFDL develop branch, and all merge conflicts are resolved

### CODE REVIEW

File: README.md
Comment (wrongkindofdoctor) [inline README.md:63]: @yihungkuo 
```
<<<<<<< HEAD

============

>>>>>>>> <commit hash>
```
Indicates that there are unresolved merge conflicts. You need to select the text you want above or below the `===========` line (or a combination of the two if there are parts of both you want to retain), and remove the `<<<<<, =====, >>>>>` lines.

File: doc/sphinx/dev_walkthrough.rst
Comment (wrongkindofdoctor) [inline doc/sphinx/dev_walkthrough.rst:17]: Resolve merge conflict

File: doc/sphinx/start_install.rst
Comment (wrongkindofdoctor) [inline doc/sphinx/start_install.rst:38]: Resolve merge conflict

File: doc/sphinx/start_install.rst
Comment (wrongkindofdoctor) [inline doc/sphinx/start_install.rst:78]: Resolve merge conflict

File: doc/sphinx/start_install.rst
Comment (wrongkindofdoctor) [inline doc/sphinx/start_install.rst:91]: Resolve merge conflict

**Review by wrongkindofdoctor — CHANGES_REQUESTED** [review]

File: README.md
Comment (jkrasting) [inline README.md:115]: Note that for CPO-funded PODs, Python 3 is required?

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (jkrasting) [inline diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:514]: Is there a need to keep the old line around?

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (jkrasting) [inline diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:590]: Is there a need to keep the old line around?

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (jkrasting) [inline diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:796]: Is there a need to keep the old line around?

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (jkrasting) [inline diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:622]: Is there a need to keep the old line around?

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (jkrasting) [inline diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:897]: Is there a need to keep the old line around?

File: README.md
Comment (yihungkuo) [reply README.md:63]: @wrongkindofdoctor and @jkrasting I am confused. I got these from the latest GFDL develop branch just a few hours ago, and these were already there. Could you confirm? And is there any thing I need to do?

File: diagnostics/convective_transition_diag/doc/convective_transition_diag.rst
Comment (jkrasting) [inline diagnostics/convective_transition_diag/doc/convective_transition_diag.rst:36]: Should we host this sample at GFDL as well?

**Review by jkrasting — CHANGES_REQUESTED** [review]
Thanks @yihungkuo - looks pretty good!  Can we work together to create some synthetic or corase resolution data for rapid testing?

I echo @wrongkindofdoctor's comments to update the merge conflicts.  Also, I noticed in some places you are keeping the older lines but commenting them out.  Is there a need to do this since this is under version control?  Also, I am not super excited about the inclusion of the Matlab files, but it's probably a lot of work to convert them to something else.

What are the changes to `src/cmip6-cmor-tables`?

Any chance you'd be willing to also provide a `settings.yaml` file as well?  Here's a free converter:  https://www.json2yaml.com/

File: diagnostics/convective_transition_diag/doc/convective_transition_diag.rst
Comment (yihungkuo) [reply diagnostics/convective_transition_diag/doc/convective_transition_diag.rst:36]: That makes sense to me.

File: README.md
Comment (yihungkuo) [reply README.md]: Fixed.

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (yihungkuo) [reply diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:590]: I would like to uncomment this line, but the newer matplotlib version doesn't like this line. Haven't figure it out. Keep it for now.

File: README.md
Comment (wrongkindofdoctor) [reply README.md]: These lines are not in the develop branch of NOAA-GFDL fork, but they are in your fork, which means that the conflicts occurred when you updated your copy of the develop branch.

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (yihungkuo) [reply diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:622]: Deleted.

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (yihungkuo) [reply diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:897]: Keep it for now before figuring out how to uncomment the line.

File: README.md
Comment (yihungkuo) [reply README.md]: Got it. Fixed.

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (yihungkuo) [reply diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py:514]: Deleted.

**Comment by yihungkuo** [comment]
> What are the changes to `src/cmip6-cmor-tables`?
@jkrasting and @wrongkindofdoctor I did not change the content of this directory. Probably something went wrong when I got the latest GFDL develop branch. Can anyone of you reject changes made to this directory?
Let's chat about this in the mini-leads telecon.

**Comment by yihungkuo** [comment]
> Any chance you'd be willing to also provide a `settings.yaml` file as well? Here's a free converter: https://www.json2yaml.com/

@jkrasting `settings.yaml` file added under diagnostics/convective_transition_diag/.

File: diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py
Comment (yihungkuo) [reply diagnostics/convective_transition_diag/convecTransCriticalCollapse_util.py]: Deleted.

**Comment by yihungkuo** [comment]
> Thanks @yihungkuo - looks pretty good! Can we work together to create some synthetic or corase resolution data for rapid testing?

@jkrasting Cannot get it working on my laptop. Need some help. Chat later.

**Comment by jkrasting** [comment]
@yihungkuo - whenever you are happy with this, please request a re-review through GitHub for the corrections.

**Review by jkrasting — APPROVED** [review]
Looks good to me @yihungkuo - thank you for the updates

**Review by wrongkindofdoctor — DISMISSED** [review]

**Comment by tsjackson-noaa** [comment]
@wrongkindofdoctor, @jkrasting: apologies for clearing your previous reviews by changing the base branch for the merge -- didn't realize there was no undo for this.

@yihungkuo: thanks for going through the code review checklist, and apologies for the delay in getting these changes merged in. POD code looks good, but I have reservations about making changes to the python3_base conda environment this close to the January deadline for other POD contributors.

Jupyterlab shouldn't be included in a "production" conda environment: although it's essential for development and debugging, we want the conda environment to contain only the dependencies needed to run the POD's code. This is in response to concerns (raised by Andrew and others) over the size of files conda installs.

I propose to merge in the PR so that the code can be included in the pre-beta-3 testing, but to temporarily revert changes to the conda environments (ie putting all changes into src/conda/env_convective_transition_diag.yml).

**Review by tsjackson-noaa — COMMENTED** [review]
Approved, with reservations described in https://github.com/NOAA-GFDL/MDTF-diagnostics/pull/84#issuecomment-756587672.

**Review by tsjackson-noaa — DISMISSED** [review]
Approved, with reservations described in https://github.com/NOAA-GFDL/MDTF-diagnostics/pull/84#issuecomment-756587672.

**Review by wrongkindofdoctor — APPROVED** [review]

**Comment by wrongkindofdoctor** [comment]
@tsjackson-noaa Moving the jupyterlab dependency to the env_convective_transition_diag.yml file sounds preferable. To clarify, do you mean that you will modify the conda environment files after merging this PR, push another alpha testing tag (after possibly merging other updates), then move the jupyterlab settings back to the python3_base conda environment? Or, will the jupyterlab settings remain in the env_convective_transition_diag.yml file until further notice (i.e., we save changing python3_base conda for another release or minor update)?

File: src/conda/env_python3_base.yml
Comment (jkrasting) [inline src/conda/env_python3_base.yml:19]:
```suggestion
```

Per comments by @tsjackson-noaa and @wrongkindofdoctor.  I agree.

**Review by jkrasting — CHANGES_REQUESTED** [review]
Approved here as well, pending the Jupyter/environment issue.

**Comment by tsjackson-noaa** [comment]
Asking for confirmation from @yihungkuo to be 100% sure: the script only uses either tave or qsat_int but never both, depending on the user setting of `BULK_TROPOSPHERIC_TEMPERATURE_VAR`. If this data isn't supplied by the model, the script can generate either from 4D temp (in convecTransBasic_calcTaveQsatInt in convecTransBasic_util.py). 
The POD's current [settings file](https://github.com/yihungkuo/MDTF-diagnostics/blob/develop/diagnostics/convective_transition_diag/settings.jsonc) lists specific humidity as an alternate variable, but this isn't referenced in the code itself, so I wanted to make sure it wasn't used.

**Comment by yihungkuo** [comment]
> Asking for confirmation from @yihungkuo to be 100% sure: the script only uses either tave or qsat_int but never both, depending on the user setting of `BULK_TROPOSPHERIC_TEMPERATURE_VAR`. If this data isn't supplied by the model, the script can generate either from 4D temp (in convecTransBasic_calcTaveQsatInt in convecTransBasic_util.py).
> The POD's current [settings file](https://github.com/yihungkuo/MDTF-diagnostics/blob/develop/diagnostics/convective_transition_diag/settings.jsonc) lists specific humidity as an alternate variable, but this isn't referenced in the code itself, so I wanted to make sure it wasn't used.
@tsjackson-noaa  A minor correction. The POD uses two kinds of variables, one for water vapor and one for temperature. 

For water vapor, it currently uses prw (lon-lat-time). Although prw can be computed from hus (lon-lat-pres-time) (and possibly requires ps and huss at surface), the POD does not provide this functionality. 

For temperature, the POD uses both tave and qsat_int when `BULK_TROPOSPHERIC_TEMPERATURE_VAR=1`, and uses only qsat_int when it's set to 2. So the POD will check the availability of both variables. If these do not exist, the POD will look for ta (lon-lat-press-time) and compute tave and qsat_int.

File: src/conda/env_python3_base.yml
Comment (yihungkuo) [reply src/conda/env_python3_base.yml]: @jkrasting, @tsjackson-noaa, @wrongkindofdoctor Jupyter removed from python3 base. 

While I was working on it, I noticed that the current version of git-intro documentation may not be ideal. It contains a lot of stuff, and I had difficulty finding the commands I needed (in the right order; e.g., `git commit -a` then `git push origin`). In the end, I went back to the older version @tsjackson-noaa wrote (last summer?) and found what I needed. 

I would like to suggest re-organizing/writing the git-intro so that users can follow it more easily (like @tsjackson-noaa's original version). Should I bring it up in this week's telecon or start an issue?

**Review by tsjackson-noaa — APPROVED** [review]

**Review by wrongkindofdoctor — APPROVED** [review]

**Comment by tsjackson-noaa** [comment]
Closing this PR since its code has been accepted as part of the [v3.0 beta 3](https://github.com/NOAA-GFDL/MDTF-diagnostics/releases/tag/v3.0-beta.3) release. See PRs NOAA-GFDL/MDTF-diagnostics#147, NOAA-GFDL/MDTF-diagnostics#149.

**Pull request closed without merging**
